packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1575,6 +1583,7 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
         volumeSize: ebsEnabled ? volumeSize : undefined,
         volumeType: ebsEnabled ? volumeType : undefined,
         iops: ebsEnabled ? props.ebs?.iops : undefined,
+        throughput: ebsEnabled ? props.ebs?.throughput : undefined,
```

Comment:
It doesn't look like we have any check on the volumeType. It says iops and
throughput are only supported for iops/gp3 types, but we only check if it's ebs
type?

Do you know what happens if `throughput` is defined for `gp2` volumes? Does it
just ignore that property or does it throw an error?

I think we need something like what we have in the `volume` construct
https://github.com/aws/aws-cdk/blob/53d24f1a1357410746052bbf25b5b89b985f7b80/packages/aws-cdk-lib/aws-ec2/lib/volume.ts#L704-L771

Reply:
I did think about checks specification checks, the other option for `iops` didn't any checks defined so I didn't it for the `throughput` option`. I'll get that sorted in the next commit :)

Reply:
Added a bunch of tests for you @corymhall